bitcoin_executor/sources/tx.move

```diff
@@ -1,25 +1,212 @@
 module bitcoin_executor::tx;
-use bitcoin_executor::interpreter::run;
 
+use bitcoin_executor::interpreter::run;
+use bitcoin_executor::reader::Reader;
+use bitcoin_executor::utils::u64_to_varint_bytes;
 
-/// Inputs in btc transaction
+/// Input in btc transaction
 public struct Input has copy, drop {
-    data: vector<u8>
+    tx_id: vector<u8>,
+    vout: vector<u8>,
+    script_sig: vector<u8>,
+    sequence: vector<u8>
+}
+
+/// Output in btc transaction
+public struct Output has copy, drop {
+    amount: vector<u8>,
+    script_pubkey: vector<u8>
+}
+
+public struct Witness has copy, drop{
```

Comment:
something wrong here. I think it should be just `vector<u8>`

Reply:
we are using it in Transaction struct as `vector<Witness` we end up with `vector<vector<vector<u8>>>`

Reply:
Yes, each input has one witness corresponding to this. So this is the vector of Witness for the transaction.